.github/ci_templates/steps.yml

```diff
@@ -1,95 +1,125 @@
 {!{ define "checkout_step" }!}
+# <template: checkout_step>
 - name: Checkout sources
   uses: {!{ index (ds "actions") "actions/checkout" }!}
+  {!{ if coll.Has . "pullRequestRefField" -}!}
+  with:
+    ref: ${{ {!{ .pullRequestRefField }!} }}
+  {!{- end }!}
+# </template: checkout_step>
 {!{- end -}!}
 
 {!{ define "checkout_full_step" }!}
+# <template: checkout_full_step>
 - name: Checkout sources
   uses: {!{ index (ds "actions") "actions/checkout" }!}
   with:
     fetch-depth: 0
+  {!{- if coll.Has . "pullRequestRefField" }!}
+    ref: ${{ {!{ .pullRequestRefField }!} }}
+  {!{- end }!}
+# </template: checkout_full_step>
 {!{- end -}!}
 
 {!{ define "checkout_from_event_ref_step" }!}
+# <template: checkout_from_event_ref_step>
 - name: Checkout sources
   uses: {!{ index (ds "actions") "actions/checkout" }!}
   with:
     ref: ${{ github.event.ref }}
     fetch-depth: 0
-{!{- end -}!}
+# </template: checkout_from_event_ref_step>
+{!{- end }!}
```

Comment:
```suggestion
{!{- end -}!}
```

Reply:
Unfortunately it creates malformed YAML.

```yaml
      # <template: checkout_full_step>
      - name: Checkout sources
        uses: actions/checkout@v2.4.0
        with:
          fetch-depth: 0
          ref: ${{ needs.pull_request_info.outputs.ref }}# </template: checkout_full_step>
```